### ISSUE TITLE
Dynamic sizing help/fix

### ISSUE THREAD

**Issue by adam-sroka** [salim-lachdhaf/searchable_dropdown#407]
Hi and thanks a lot for the awesome package 💙!

I was wondering if you could point me in the right direction to fix dynamically sized dropdowns — i.e. I'd like the vertical height of the dropdown list (menu) to dynamically match the number of items in it.

Previously, I used to use `maxHeight` to hardcode this, but if I understand correctly, dynamic sizing should be possible in the new 4.0.0 version:

> this will be available in the next version (dynamic height).
for now you can use `maxHeight`

_Originally posted by @salim-lachdhaf in https://github.com/salim-lachdhaf/searchable_dropdown/issues/395#issuecomment-1120263730_

I haven't found any examples or documentation for dynamic sizing, but looking at the api, it seems that this should be controlled by something in [`popupProps`](https://github.com/salim-lachdhaf/searchable_dropdown/blob/master/lib/src/properties/popup_props.dart), or more specifically, the `fit` attribute:

```dart
  ///fit height depending on nb of result or keep height fix.
  final FlexFit fit;
```

I'm not too familiar with the `FlexFit` enum, but I've tried both `FlexFit.tight` and `FlexFit.loose` and I always _either_ get a dropdown that scrolls (is not tall enough) _or_ a dropdown that has blank space (is too tall, if I set `menuProps: MenuProps(constraints: BoxConstraints(maxHeight: "some_larger_than_needed_value"))`:
Short:
![short](https://user-images.githubusercontent.com/45038919/168642675-d1fa5358-fe39-4162-b66b-d02aafca3149.png)
Tall:
![tall](https://user-images.githubusercontent.com/45038919/168642728-db8b2920-c33f-4b77-b92c-5e04c4661984.png)
The code snippet for this is:
```dart
const List<String> recallDayTypes = [
      'Normal day',
      'Sick day',
      'Fasting day',
      'Festival/religious day',
      'Parties/functions day',
      'Visitors/relatives',
      'Other'
    ];
//...
DropdownSearch<String>(
            popupProps: const PopupProps.menu(
              showSelectedItems: true,
              fit: FlexFit.tight,
              // The "correct" height for this dropdown of `recallDayTypes` would be 395.00
              // menuProps:
              //     MenuProps(constraints: BoxConstraints(maxHeight: 595.0))
            ),
            dropdownSearchDecoration: InputDecoration(
                icon: const Icon(Icons.food_bank_outlined),
                labelText: "Recall Day",
                helperText: 'Type of the recall day',
            items: recallDayTypes,
            onChanged: (String? recallDayType) => context
                .read<CollectionBloc>()
                .add(RecallDayChanged(recallDay: recallDayType ?? '')),
            selectedItem: state.gibsonsForm.recallDay)
//...
```

Thank you very much for any help! 😊

**Comment by salim-lachdhaf** [comment]
two step to get your need:
1- use `fit` to `FlexFit.loose`
2- play with `constraints`  (try to set only `minHeight`)

let me now if you het it working.

**Comment by adam-sroka** [comment]
> two steps to get your need:
> 1- use `fit` to `FlexFit.loose`
> 2- play with `constraints` (try to set only `minHeight`)
> 
> let me now if you het it working.

Thanks a lot for the fast response, @salim-lachdhaf, I really appreciate it!

Unfortunately, I do not seem to get it working. I modified the flutter "create new project" example to illustrate what I want to do (full code is below), and here is a screen recording of me playing around with different options while trying to get dynamic sizing to work:

https://user-images.githubusercontent.com/45038919/168694042-bb0bd5da-c57b-4ce4-9f4e-86bcc240b20f.mp4

I start at the beginning with hardcoded `maxHeight` of 395.00 : `menuProps:MenuProps(constraints: BoxConstraints(maxHeight: 395.0))` — this is the height that I am actually trying to achieve, without needing to hardcode it!

I then try setting the `fit` to `FlexFit.loose` (instead of the default `tight`), but this has no effect. If I then change `maxHeight` to `minHeight: "anyValue"`, the dropdown just expands vertically and fills the whole page (i.e. is taller than the height of the `items` that I want to dynamically match). This happens regardless of what value I set `minHeight` to.

Even when changing the `fit` to `FlexFit.tight` (by commenting out `FlexFit.loose`, the behaviour is the same, i.e. setting `minHeight` to any value always results in the dropdown filling the whole page, regardless of what `fit` is set to.

Not shown in the video, but setting both `minHeight` and `maxHeight` is the same as setting just `maxHeight`, i.e. the dropdown fills to expand until `maxHeight`.

Thus, so far, I cannot find any other way of making the dropdown be as tall as its `items`, apart from manually setting the `maxHeight` to the correct height of `items`.

I'd appreciate any more help 😊.

Here is the full code for reference:
```dart
import 'package:flutter/material.dart';
import 'package:dropdown_search/dropdown_search.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;
  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String? _selectedRecallDayType;
  void _selectRecallDayType(String? selectedRecallDayType) {
    setState(() {
      _selectedRecallDayType = selectedRecallDayType;
    });
  }

  @override
  Widget build(BuildContext context) {
    const List<String> recallDayTypes = [
      'Normal day',
      'Sick day',
      'Fasting day',
      'Festival/religious day',
      'Parties/functions day',
      'Visitors/relatives',
      'Other'
    ];
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            DropdownSearch<String>(
                popupProps: const PopupProps.menu(
                    showSelectedItems: true,
                    // fit: FlexFit.loose,
                    // The "correct" height for this dropdown would be 395.00
                    menuProps:
                        MenuProps(constraints: BoxConstraints(maxHeight: 395.0))
                    //
                    ),
                dropdownSearchDecoration: const InputDecoration(
                  icon: Icon(Icons.food_bank_outlined),
                  labelText: "Recall Day",
                  helperText: 'Type of the recall day',
                ),
                items: recallDayTypes,
                onChanged: (String? recallDayType) =>
                    _selectRecallDayType(recallDayType),
                selectedItem: _selectedRecallDayType)
          ],
        ),
      ),
    );
  }
}
```

**Comment by salim-lachdhaf** [comment]
use this two params :
```
fit: FlexFit.loose,
menuProps: MenuProps(
  constraints: BoxConstraints.tightFor(),
)
```

**Comment by salim-lachdhaf** [comment]
i confirm that in the single mode , there is a bug, i'm correcting it.

**Comment by salim-lachdhaf** [comment]
fixed in v4.0.1.

**Comment by adam-sroka** [comment]
Yaay, it works! Thanks so much @salim-lachdhaf 😊!
